docs/DeveloperGuide.md

```diff
@@ -255,30 +255,55 @@ _{Explain here how the data archiving feature will be implemented}_
+|----------|-------------------------|------------------------------------------------------------|----------------------------------------------------------------------------|
+| `* * *`  | new user                | see usage instructions                                     | refer to instructions when I forget how to use the App                     |
+| `* * *`  | user                    | add a new contact                                          |                                                                            |
+| `* * *`  | user                    | delete a contact                                           | remove entries that I no longer need                                       |
+| `* * *`  | user                    | view all my contacts                                       | keep track of all my contacts                                              |
+| `* *`    | user                    | edit a contact                                             | update it or correct mistakes                                              |
+| `* *`    | user                    | clear all my contacts                                      | save time deleting them one by one                                         |
+| `* *`    | user                    | create a new tag                                           | categorise my contacts                                                     |
+| `* *`    | user                    | delete a tag                                               | remove redundant tags                                                      |
+| `* *`    | user                    | edit a tag                                                 | provide better naming                                                      |
+| `* *`    | user                    | tag a contact                                              | add it to a category                                                       |
+| `* *`    | user                    | untag a contact                                            | remove it from a category                                                  |
+| `* *`    | user                    | search using a name, phone number, email or address        | find a contact easily without reading through the list                     |
+| `* *`    | user                    | search with a tag                                          | find groups of contacts that share a common tag                            |
+| `* *`    | user                    | search with multiple tags                                  | narrow my search results to only contacts that have all the specified tags |
+| `* *`    | user                    | search with multiple tags                                  | broaden my result results to contacts that have any of the specified tags  |
```

Comment:
These two user stories sound a bit contradicting to me, maybe just the first one will be good enough, what do you think?

Reply:
These are two different functions for the improved search feature
1. matches contacts with **all** the specified tags
2. matches contacts with **any** of the specified tags (any = can be 1, can be multiple, can be all the tags)

Reply:
okay now it makes sense, I didn't see the all and any. Thanks for clarifying.